projects/meta_cl/networks/dendritic_networks.py

```diff
@@ -184,6 +182,116 @@ def conv_block(cls, in_channels, out_channels, kernel_size, stride, padding):
                 in_channels=in_channels, out_channels=out_channels,
                 kernel_size=kernel_size, stride=stride, padding=padding
             ),
+            nn.InstanceNorm2d(num_features=out_channels, affine=True),
+            nn.ReLU(),
+        ]
+
+    def apply_dendrites(self, y, dendrite_activations):
+        """
+        Apply dendrites as a gating mechanism.
+
+        TODO: Decide best method to apply dendrites.
+        """
+        # # Multiple by the sigmoid of the max along each segment.
+        # return y * torch.sigmoid(dendrite_activations.max(dim=2).values)
+
+        inds = dendrite_activations.abs().max(dim=2).indices
+        inds = inds.unsqueeze(dim=2)
+        dendrite_activations = torch.gather(dendrite_activations, dim=2, index=inds)
+        dendrite_activations = dendrite_activations.squeeze(dim=2)
+        dendrite_activations = torch.sigmoid(dendrite_activations)
+        return y * dendrite_activations
+
+    def forward(self, x):
+
+        mod = self.modulation(x)
+        pred = self.prediction(x)
+
+        dendrite_activations = self.segments(mod)
+        out = self.apply_dendrites(pred, dendrite_activations)
+
+        out = self.classifier(out)
+
+        return out
+
+
+class CloserToANMLDendriticNetwork(nn.Module):
+    """
+    An iteration of a dendritic network, based closely on `ANML`_. (The conv + max-pool
+    layers are identical). It's similar to ANMLDendriticNetwork, but does not include a
+    linear layer atop the modulation network. This makes it more similar to ANML's
+    original network. In fact, with `num_segments=1, dendrite_sparsity=0`, they should
+    be identical.
+
+    .. _ANML: nupic.research/projects/meta_cl/networks/anml_networks.py
+
+    Input shape expected is 3 x 28 x 28 (C x W x H) but an adaptive average pooling
+    layer helps accept other heights and width.
+
+    With default parameters and `num_classes-963`, it uses 5,132,832 weights-on
```

Comment:
Just double checking, is the ANML network output size 963, exact number of classes as in the background set? I had it in my head it was 1000.

Reply:
The ANML repo used 1000 but we used 963. In my experiments, this didn't seem to make a difference, so I left it there.

Reply:
Sure, I just thought it was important to mention since you were tracking down the last few points in difference in accuracy between the two repos.